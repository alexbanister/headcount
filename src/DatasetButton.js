import React from 'react';

const DatasetButton = ( {dataSet} ) => {
  return (
    <option value={dataSet}>
      {dataSet}
    </option>
  );
};

export default DatasetButton;
