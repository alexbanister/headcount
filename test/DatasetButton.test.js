import React from 'react';
import { shallow } from 'enzyme';
import DatasetButton from '../src/DatasetButton';
import DistrictRepository from '../src/helper.js';
import kinderData from '../data/kindergartners_in_full_day_program.js';

describe('DatasetButton', () => {
  const district = new DistrictRepository(kinderData);

  it('should create an instance of a card container', () => {
    const wrapper = shallow(<DatasetButton
      key={district.districtData.COLORADO.location}
      dataSet={district.districtData.COLORADO.location}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
