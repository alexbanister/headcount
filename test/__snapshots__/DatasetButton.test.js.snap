// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetButton should create an instance of a card container 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DatasetButton
    dataSet="COLORADO"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": "COLORADO",
      "value": "COLORADO",
    },
    "ref": null,
    "rendered": "COLORADO",
    "type": "option",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "COLORADO",
        "value": "COLORADO",
      },
      "ref": null,
      "rendered": "COLORADO",
      "type": "option",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
